frb_codegen/src/parser/mod.rs

```diff
@@ -488,3 +560,60 @@ fn extract_metadata(attrs: &[Attribute]) -> Vec<IrDartAnnotation> {
 fn type_to_string(ty: &Type) -> String {
     quote!(#ty).to_string().replace(' ', "")
 }
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use syn::{parse_str, Type};
+
+    use crate::parser::topo_resolve;
+
+    #[test]
+    fn test_topo_resolve1() {
+        let mut input = HashMap::new();
+        let mut expect = HashMap::new();
+        input.insert("id".to_string(), parse_str::<Type>("i32").unwrap());
+        input.insert("UserId".to_string(), parse_str::<Type>("id").unwrap());
+        let input = input;
+        expect.insert("id".to_string(), parse_str::<Type>("i32").unwrap());
+        expect.insert("UserId".to_string(), parse_str::<Type>("i32").unwrap());
+        let output = topo_resolve(input);
+        assert_eq!(output, expect);
+    }
+    #[test]
+    fn test_topo_resolve2() {
```

Comment:
uber-nit: would be great if each test has a good name instead of 1 2 3 (if possible)